govtool/packages/wallet-connector/components/ChooseWalletModal.tsx

```diff
@@ -0,0 +1,140 @@
+import { useMemo } from "react";
+import { Box, Link, Modal, Typography } from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+
+import { WalletButton } from "./WalletButton";
+import { ChooseWalletModalProps, WalletInfo } from "./types";
+
+export const ChooseWalletModal = ({
+  handleClose,
+  isOpen,
+  isWalletEnableLoading,
+  onClickShowSupportedWallets,
+  onClickWalletButton,
+}: ChooseWalletModalProps) => {
+  const walletOptions = useMemo(() => {
+    if (!window.cardano) return [];
+    const keys = Object.keys(window.cardano);
+    let resultWallets: WalletInfo[] = [];
+    keys.forEach((k: string) => {
+      const { icon, name, supportedExtensions } = window.cardano[k];
+      if (icon && name && supportedExtensions) {
+        const isNameDuplicate = resultWallets.some(
+          (wallet) => wallet.name === name
+        );
+
+        const isCip95Available = Boolean(
+          supportedExtensions?.find((i) => i.cip === 95)
+        );
+
+        if (!isNameDuplicate && isCip95Available) {
+          resultWallets.push({
+            icon,
+            label: k,
+            name,
+          });
+        }
+      }
+    });
+
+    return resultWallets;
+  }, [window]);
+
+  return (
+    <Modal onClose={handleClose} open={isOpen}>
+      <Box sx={style.container}>
+        <CloseIcon
+          data-testid="close-modal-button"
+          onClick={handleClose}
+          sx={style.closeButton}
+        />
+        <Box>
+          <Typography sx={style.header}>Connect your Wallet</Typography>
+          <Typography sx={style.description}>
+            Choose the wallet you want to connect with
+          </Typography>
+          <Box sx={style.buttonsContainer}>
+            {!walletOptions.length ? (
+              <Typography sx={style.emptyState}>
+                You don't have wallets to connect, install a wallet and refresh
+                the page and try again
+              </Typography>
+            ) : (
+              walletOptions.map(({ icon, label, name }) => (
+                <WalletButton
+                  onClick={() => onClickWalletButton(name)}
+                  isLoading={isWalletEnableLoading}
+                  dataTestId={`${name}-wallet-button`}
+                  key={name}
+                  icon={icon}
+                  label={label}
+                  name={name}
+                />
+              ))
+            )}
+          </Box>
+          <Typography sx={style.information}>
```

Comment:
that block should be conditionally rendered based on the `onClickShowSupportedWallets` hence, `onClickShowSupportedWallets` should be optional